src/files.ts

```diff
@@ -134,11 +134,31 @@ export function parseFileIgnores(path: string): string[] {
 
 export function getGlobPatterns(supportedFiles: SupportedFiles): string[] {
   return [
-    ...supportedFiles.extensions.map(e => `*${e}`),
+    ...supportedFiles.extensions.map(e => `*${generateAllCaseGlobPattern(e)}`),
     ...supportedFiles.configFiles.filter(e => !EXCLUDED_NAMES.includes(e)),
   ];
 }
 
+// Generates glob patterns for case-insensitive file extension matching.
+// E.g. *.[jJ][sS] for matching .js files without case-sensitivity.
+function generateAllCaseGlobPattern(fileExtension: string): string {
+  const chars = Array.from(fileExtension);
+  const p = chars.reduce((pattern: string[], extensionChar, i) => {
+    if (i == 0) {
+      // first char is always '.', no need to generate multiple cases for file extension character
+      if (extensionChar != '.') {
+        throw new Error('Unexpected file extension pattern when constructing glob patterns.');
```

Comment:
req/q: This will cause every analysis to fail if we ever return an entry that doesn't start with `.`, right? Maybe we should remove the special handling for the `.` character, uppercase what we can, and ignore what we cannot uppercase?

Reply:
Correct. The problem with ignoring a missing `.` is that we won't catch the problem in any way, keeping us running in a blind-mode. 

Ideally this validation should be somewhere on the backend actually, and client-side/analysis shouldn't be impacted. I'll convert this to `console.log` for general visibility.